requirements/prod.txt

```diff
@@ -28,7 +28,12 @@ scalecodec==1.2.7
 shtab==1.6.5
 substrate-interface==1.7.5
 termcolor
-torch>=1.13.1
+torch @ https://download.pytorch.org/whl/cpu-cxx11-abi/torch-2.2.0%2Bcpu.cxx11.abi-cp312-cp312-linux_x86_64.whl ; sys_platform == 'linux' and platform_machine == 'x86_64' and python_version == '3.12'
+torch @ https://download.pytorch.org/whl/cpu-cxx11-abi/torch-2.2.1%2Bcpu.cxx11.abi-cp311-cp311-linux_x86_64.whl ; sys_platform == 'linux' and platform_machine == 'x86_64' and python_version == '3.11'
+torch @ https://download.pytorch.org/whl/cpu-cxx11-abi/torch-2.0.1%2Bcpu.cxx11.abi-cp310-cp310-linux_x86_64.whl ; sys_platform == 'linux' and platform_machine == 'x86_64' and python_version == '3.10'
+torch @ https://download.pytorch.org/whl/cpu-cxx11-abi/torch-2.0.1%2Bcpu.cxx11.abi-cp39-cp39-linux_x86_64.whl ; sys_platform == 'linux' and platform_machine == 'x86_64' and python_version == '3.9'
+torch @ https://download.pytorch.org/whl/cpu-cxx11-abi/torch-2.0.1%2Bcpu.cxx11.abi-cp38-cp38-linux_x86_64.whl ; sys_platform == 'linux' and platform_machine == 'x86_64' and python_version == '3.8'
+torch>=1.13.1 ; sys_platform != 'linux' or platform_machine != 'x86_64'
```

Comment:
not sure if all these python versions need to be supported

Reply:
I think freezing version like this is not very good. I'd just specify index url.

As I understand, one cannot specify index url per-package in `requirements.txt`, so I would do something like this:

```
# requirements-prod.txt
package1
package2
-r requirements-torch.txt
```
and
```
# requirements-torch.txt
--index-url=https://download.pytorch.org/whl/cpu
torch>=1.13.1
```
(https://stackoverflow.com/a/11704396/1935381)

No hard pinning. Will it work? Wdyt?

Reply:
that doesn't work as they are parsing the requirements.txt file https://github.com/andreea-popescu-reef/bittensor/blob/master/setup.py#L36 to obtain install_requires and I don't think that supports index url

Reply:
This is also concerning from a maintainability perspective. 

Yes it works now, but how many headaches could this approach cause moving forward?

Reply:
This also assumes that users of bittensor will have to install cuda directly if they require a GPU, yes?

This may be unacceptable to force on end users if they are not developers or sophisticated

Reply:
Also, please follow convention in `contrib` (have a read) and branch off from staging.

Thanks!

Reply:
CLI users don't need GPUs.

From packaging perspective I don't think using the same repository for CLI and a library which includes the sdk that subnet validator and miners use, is the best possible choice, and then we can have variants like:
- `bittensor` for cli
- `bittensor[cpu]` for those subnets that don't rely on gpu (sn7, sn12, sn18, sn21 and probably many others)
- `bittensor[gpu]` that the GPU-based subnets can inherit from

But lets not make everybody download extra 5GB of stuff they don't use.

Reply:
This is reasonable. I think having separate sub-packages `cpu/gpu` is a simultaneous requirement for this PR then